check-shortlog/action.yml

```diff
@@ -0,0 +1,54 @@
+name: Check Shortlog
+description: |
+  Check the output of git shortlog -s matches the list of contributors in the
+  CONTRIBUTING.md file.
+
+inputs:
+  contributing_file:
+    description: Path to contributing file.
+    default: CONTRIBUTING.md
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Get known contributors from contributing file
+      shell: bash
+      run: |
+        echo '::group::Get known contributors from contributing file'
+        sed \
+            -n \
+            -e 's/ - //' \
+            -e 's/^(//' -e 's/)$//' \
+            -e '/start-shortlog/,${p;/end-shortlog/q}' \
+            "${{ inputs.contributing_file }}" \
+            | head -n -1 \
+            | tail -n +2 \
+            | sort \
+            > contributors
+        cat contributors
+        echo "::endgroup::"
+
+    - name: List commit authors
+      shell: bash
+      run: |
+        echo '::group::List commit authors'
+        git shortlog -s "${{ github.ref }}" \
```

Comment:
Have to explicitly provide the ref when working non-interactively but not otherwise?!

Whatever.

Reply:
What about `git shortlog -s HEAD`?

Reply:
Tried, it didn't seem to like it.

Reply:
`github.sha` seems to work non-interactively for all situations, merge commits, workflow dispatch, locally.